pallets/passkey/src/lib.rs

```diff
@@ -127,53 +131,115 @@ pub mod module {
 	impl<T: Config> ValidateUnsigned for Pallet<T> {
 		type Call = Call<T>;
 		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
-			Self::validate_signatures(call)?;
-			Ok(ValidTransaction::default())
+			let payload = Self::filter_valid_calls(&call)?;
+			Self::validate_signatures(&payload)?;
+			Self::validate_nonce(&payload)
+		}
+
+		fn pre_dispatch(call: &Self::Call) -> Result<(), TransactionValidityError> {
+			Self::validate_unsigned(TransactionSource::InBlock, call)?;
+
+			let payload = Self::filter_valid_calls(&call)?;
+			Self::pre_dispatch_nonce(&payload)
 		}
 	}
+}
 
-	impl<T: Config> Pallet<T> {
-		fn validate_signatures(call: &Call<T>) -> TransactionValidity {
-			match call {
-				Call::proxy { payload } => {
-					let signed_data = payload.passkey_public_key;
-					let signature = payload.passkey_call.account_ownership_proof.clone();
-					let signer = &payload.passkey_call.account_id;
-					match Self::check_account_signature(signer, &signed_data.into(), &signature) {
-						Ok(_) => Ok(ValidTransaction::default()),
-						Err(_e) => InvalidTransaction::BadSigner.into(),
-					}
-				},
-				_ => InvalidTransaction::Call.into(),
-			}
+impl<T: Config> Pallet<T> {
+	fn filter_valid_calls(call: &Call<T>) -> Result<PasskeyPayload<T>, TransactionValidityError> {
+		match call {
+			Call::proxy { payload }
+				if T::PasskeyCallFilter::contains(&payload.clone().passkey_call.call) =>
+				return Ok(payload.clone()),
+			_ => return Err(InvalidTransaction::Call.into()),
+		}
+	}
+
+	fn validate_nonce(payload: &PasskeyPayload<T>) -> TransactionValidity {
+		let who = &payload.passkey_call.account_id.clone();
```

Comment:
I believe we could reuse the nonce validation from the extension and call that function instead of duplicating the code here. While duplicating code can be acceptable in some cases, it would be beneficial to keep both validations always up to date.

Additionally, it might be a good idea to figure out how to integrate this approach, as it seems we will need to draw from other extensions like validating and processing payments.

Reply:
I already tried that approach. After spending more than 2 hours and the code getting much more complicated than it should have been. I was not able to make it work. One of the trait constraints didn't work as I was hoping and I had to switch to this approach in the end.
 
If anybody would like to spend some time and try it they are welcome to.

Reply:
It would be great if you can make it work.

Reply:
Ok, Cool. Ill take a look.

Reply:
@aramikm Here is an example of using the extension directly. This approach passes all current tests. Please note that I have not yet explored the details of what we need from it extensively. However, at first glance, it appears to meet our requirements.

https://github.com/frequency-chain/frequency/pull/2045

Reply:
Many thanks